src/category_theory/enriched/basic.lean

```diff
@@ -0,0 +1,292 @@
+obtained by applying the functor `F : lax_monoidal_functor V W` to each hom object.
+-/
+@[nolint has_inhabited_instance unused_arguments]
+def transport_enrichment (F : lax_monoidal_functor V W) (C : Type u₁) := C
+
+instance (F : lax_monoidal_functor V W) :
+  enriched_category W (transport_enrichment F C) :=
+{ hom := λ (X Y : C), F.obj (X ⟶[V] Y),
+  id := λ (X : C), F.ε ≫ F.map (e_id V X),
+  comp := λ (X Y Z : C), F.μ _ _ ≫ F.map (e_comp V X Y Z),
+  id_comp := λ X Y, begin
+    rw [comp_tensor_id, category.assoc,
+      ←F.to_functor.map_id, F.μ_natural_assoc, F.to_functor.map_id, F.left_unitality_inv_assoc,
+      ←F.to_functor.map_comp, ←F.to_functor.map_comp, e_id_comp, F.to_functor.map_id],
+  end,
+  comp_id := λ X Y, begin
+    rw [id_tensor_comp, category.assoc,
+      ←F.to_functor.map_id, F.μ_natural_assoc, F.to_functor.map_id, F.right_unitality_inv_assoc,
+      ←F.to_functor.map_comp, ←F.to_functor.map_comp, e_comp_id, F.to_functor.map_id],
+  end,
+  assoc := λ P Q R S, begin
+    rw [comp_tensor_id, category.assoc, ←F.to_functor.map_id, F.μ_natural_assoc,
+      F.to_functor.map_id, ←F.associativity_inv_assoc, ←F.to_functor.map_comp,
+      ←F.to_functor.map_comp, e_assoc, id_tensor_comp, category.assoc, ←F.to_functor.map_id,
+      F.μ_natural_assoc, F.to_functor.map_comp],
+  end, }
+
+end
+
+/--
+A `V`-functor `F` between `V`-enriched categories
+has a `V`-morphism from `X ⟶[V] Y` to `F.obj X ⟶[V] F.obj Y`,
+satisfying the usual axioms.
+-/
+@[nolint has_inhabited_instance]
+structure enriched_functor
+  (C : Type u₁) [enriched_category V C] (D : Type u₂) [enriched_category V D] :=
+(obj : C → D)
+(map : Π X Y : C, (X ⟶[V] Y) ⟶ (obj X ⟶[V] obj Y))
+(map_id' : ∀ X : C, e_id V X ≫ map X X = e_id V (obj X) . obviously)
+(map_comp' : ∀ X Y Z : C,
+  e_comp V X Y Z ≫ map X Z = (map X Y ⊗ map Y Z) ≫ e_comp V (obj X) (obj Y) (obj Z) . obviously)
+
+restate_axiom enriched_functor.map_id'
+restate_axiom enriched_functor.map_comp'
+attribute [simp, reassoc] enriched_functor.map_id
+attribute [simp, reassoc] enriched_functor.map_comp
+
+/-- The identity enriched functor. -/
+@[simps]
+def enriched_functor.id (C : Type u₁) [enriched_category V C] : enriched_functor V C C :=
+{ obj := λ X, X,
+  map := λ X Y, 𝟙 _, }
+
+/-- Composition of enriched functors. -/
+@[simps]
+def enriched_functor.comp {C : Type u₁} {D : Type u₂} {E : Type u₃}
+  [enriched_category V C] [enriched_category V D] [enriched_category V E]
+  (F : enriched_functor V C D) (G : enriched_functor V D E) :
+  enriched_functor V C E :=
+{ obj := λ X, G.obj (F.obj X),
+  map := λ X Y, F.map _ _ ≫ G.map _ _, }
+
+section
+variables {V} [braided_category V]
+variables {D : Type u₂} [enriched_category V D]
+
+/-!
+For general `V`-enriched categories `C D`, and `V`-functors `F G`,
+it's not possible to make sense of natural transformations between `F` and `G` at all.
+
+An essential ingredient is a braiding (or symmetry) on `V`.
```

Comment:
I think this language is a little strong, as we discussed you can make a type of V-natural transformations which do form a category: it's just that this category isn't very helpful.

Reply:
Sorry, yes, I'll rewrite this whole section to conform closer to normal usage!

Reply:
Completely rewritten now.